File: cutils/src/main/java/com/carlos/cutils/view/ToastUtil.java
package com.carlos.cutils.view;

import android.content.Context;
import android.widget.Toast;

/**
 * Created by Carlos on 2018/2/27.
 */

public class ToastUtil {

    private Toast mToast;
    private Context mContext;
    private String message="";
    private int duration;

    private ToastUtil() {
    }


    public static class Builder{
        ToastUtil mToastUtil = new ToastUtil();

        public Builder(Context context){
            mToastUtil.mContext=context;
        }

        public Builder setText(String message){
            mToastUtil.message=message;
            return this;
        }

        public Builder setShortToast(){
            mToastUtil.duration=Toast.LENGTH_SHORT;
            return this;
        }

        public Builder setLongToast(){
            mToastUtil.duration=Toast.LENGTH_LONG;
            return this;
        }

        public ToastUtil build(){
            mToastUtil.setLayoutView();
            return mToastUtil;
        }

    }

    public void setLayoutView(){
        mToast=Toast.makeText(mContext,message,duration);
        mToast.show();
    }


}

File: cutils/src/main/java/com/carlos/cutils/util/ToastUtil.kt
package com.carlos.cutils.util

import android.content.Context
import android.widget.Toast

/**
 * Created by Carlos on 2018/2/27.
 */

class ToastUtil private constructor() {

    private lateinit var mToast: Toast
    private lateinit var mContext: Context
    private var message = ""
    private var duration: Int = 0


    class Builder(context: Context) {
        private var mToastUtil = ToastUtil()

        init {
            mToastUtil.mContext = context
        }

        fun setText(message: String): Builder {
            mToastUtil.message = message
            return this
        }

        fun setShortToast(): Builder {
            mToastUtil.duration = Toast.LENGTH_SHORT
            return this
        }

        fun setLongToast(): Builder {
            mToastUtil.duration = Toast.LENGTH_LONG
            return this
        }

        fun build(): ToastUtil {
            mToastUtil.setLayoutView()
            return mToastUtil
        }

    }

    fun setLayoutView() {
        mToast = Toast.makeText(mContext, message, duration)
        mToast.show()
    }


}
